src/panels/lovelace/entity-rows/hui-climate-entity-row.ts

```diff
@@ -55,7 +57,13 @@ class HuiClimateEntityRow extends LitElement implements EntityRow {
     }
 
     return html`
-      <hui-generic-entity-row .hass="${this.hass}" .config="${this._config}">
+      <hui-generic-entity-row
+        @ha-click=${this._handleTap}
```

Comment:
Because generic entity row wraps things like `ha-climate-state` , hui-generic-entity-row will fight with the embedded component for clicks/long presses. I don't think that this will result in a good experience.

Reply:
You're right. I'll explore better options

Reply:
I can tell you that this is pretty hard, if not impossible. Having DOM elements fight with child elements for events is tough.

Reply:
To be clear; if I get this working well, you're okay with this change? Just don't want to invest more time into it otherwise.

Reply:
I don't know. Sorry, I've been grinding a lot at PRs and Google Report State stuff that I don't have been able to think about it. 

Right now we have a hard rule that the left part of entity rows are for info only. So it should then only apply to that part. Not the right side. But not all entitiy rows will have it, so we can't just add it to `EntitiesCardEntityConfig` but need to look on a per-card basis if this is something that they can support. 

However, it also means that changing how the entity row looks now has extra constraints, because if we would want to add some buttons on the left, we can no longer make it clickable for other actions. 

In the end the main thing that kinda bumps me out is that the UI doesn't look actionable, so it feels weird to actually attach an action to begin with. In Material generally it's tapping a row, card or actionable widget to do something. Having a light turn off because I tap a name feels weird.

Reply:
What about applying it to only the state badge? That would make sense for just more-info as it is now as well. All rows contain a state-badge except for the special rows

Reply:
Actually, the fighting over click events is already happening, as now every entity row gets a click listener and all elements inside entity row `stopPropagation` of the click event...

The only thing that would change is that instead of default opening the more info, the user defined action would be activated.

Reply:
I'm for either method. I'll let you and @balloob decide which makes more sense; badge or row